### DISCUSSION TITLE
How to reproduce DGCNN

### DISCUSSION THREAD

**Discussion by ZhouAo-ZA** [pyg-team/pytorch_geometric#4849] [Q&A]
I want to reproduce DGCNN on the ModelNet40 dataset, are there any examples that can help me build and train it with PyG?

I tried to build a model with 4 layers of DynamicEdgeConv according to the settings of the DGCNN paper, but I couldn't get the accuracy of the paper, so I think it may be a problem with my training method or data processing.

Any help is greatly appreciated！

**Comment by rusty1s** [comment]
Did you have a look at https://github.com/AnTao97/dgcnn.pytorch? Maybe you can spot some differences in training or data pre-processing.

**Reply by ZhouAo-ZA** [reply]
Yes, I have looked at this repository, and referenced his model construction and training approach, but the accuracy rate cannot reach what is reported in the paper, only about 91% accuracy.
So I want to ask if anyone Based on PyG implementation and achieved the accuracy of the paper report.
My model structure and training settings are as follows:
```
class DGCNN_cls(torch.nn.Module):
    def __init__(self, out_channels, k=20, aggr='max'):
        super().__init__()
        self.conv1 = DynamicEdgeConv(MLP([2 * 3, 64]), k, aggr)
        self.conv2 = DynamicEdgeConv(MLP([64 * 2, 64]), k, aggr)
        self.conv3 = DynamicEdgeConv(MLP([2 * 64, 128]), k, aggr)
        self.conv4 = DynamicEdgeConv(MLP([2 * 128, 256]), k, aggr)
        self.lin1 = Linear(512, 1024)
        self.mlp = nn.Sequential(MLP([2048, 512]), nn.Dropout(0.5),
                                 MLP([512, 256]), nn.Dropout(0.5), Linear(256, 40))

    def forward(self, data):
        x,edge_index,batch=data.pos,data.edge_index,data.batch
        # pos, batch = data.pos, data.batch
        x1 = self.conv1(x, batch)
        x2 = self.conv2(x1, batch)
        x3 = self.conv3(x2, batch)
        x4 = self.conv4(x3, batch)
        out = self.lin1(torch.cat([x1, x2,x3,x4], dim=1))
        out = torch.cat([global_max_pool(out, batch),global_mean_pool(out,batch)],dim=1)
        out = self.mlp(out)
        return F.log_softmax(out, dim=1)
device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
model = DGCNN_cls(train_dataset.num_classes, k=20).to(device)
print("Let's use", torch.cuda.device_count(), "GPUs!")
optimizer = torch.optim.Adam(model.parameters(), lr=0.001)
scheduler = torch.optim.lr_scheduler.CosineAnnealingLR(optimizer, 250, eta_min=1e-3)
```
